packages/amplify-graphql-relational-transformer/src/graphql-many-to-many-transformer.ts

```diff
@@ -0,0 +1,208 @@
+import { DirectiveWrapper, InvalidDirectiveError, TransformerPluginBase } from '@aws-amplify/graphql-transformer-core';
+import {
+  TransformerContextProvider,
+  TransformerPrepareStepContextProvider,
+  TransformerSchemaVisitStepContextProvider,
+  TransformerTransformSchemaStepContextProvider,
+  TransformerValidationStepContextProvider,
+} from '@aws-amplify/graphql-transformer-interfaces';
+import { DirectiveNode, FieldDefinitionNode, InterfaceTypeDefinitionNode, ObjectTypeDefinitionNode } from 'graphql';
+import {
+  blankObject,
+  getBaseType,
+  graphqlName,
+  isListType,
+  makeArgument,
+  makeDirective,
+  makeField,
+  makeNamedType,
+  makeValueNode,
+  toUpper,
+  wrapNonNull,
+} from 'graphql-transformer-common';
+import { ManyToManyDirectiveConfiguration, ManyToManyRelation } from './types';
+import { validateModelDirective } from './utils';
+import { makeQueryConnectionWithKeyResolver, updateTableForConnection } from './resolvers';
+import { ensureHasManyConnectionField, extendTypeWithConnection, getPartitionKeyField } from './schema';
+import { ModelTransformer } from '@aws-amplify/graphql-model-transformer';
+import { IndexTransformer } from '@aws-amplify/graphql-index-transformer';
+import { HasOneTransformer } from './graphql-has-one-transformer';
+
+const directiveName = 'manyToMany';
+const defaultLimit = 100;
+const directiveDefinition = `
+  directive @${directiveName}(relationName: String!, limit: Int = ${defaultLimit}) on FIELD_DEFINITION
+`;
+
+export class ManyToManyTransformer extends TransformerPluginBase {
+  private relationMap = new Map<string, ManyToManyRelation>();
+  private directiveList: ManyToManyDirectiveConfiguration[] = [];
+  private modelTransformer: ModelTransformer;
+  private indexTransformer: IndexTransformer;
+  private hasOneTransformer: HasOneTransformer;
+
+  constructor(modelTransformer: ModelTransformer, indexTransformer: IndexTransformer, hasOneTransformer: HasOneTransformer) {
+    super('amplify-many-to-many-transformer', directiveDefinition);
+    this.modelTransformer = modelTransformer;
+    this.indexTransformer = indexTransformer;
+    this.hasOneTransformer = hasOneTransformer;
+  }
+
+  field = (
+    parent: ObjectTypeDefinitionNode | InterfaceTypeDefinitionNode,
+    definition: FieldDefinitionNode,
+    directive: DirectiveNode,
+    context: TransformerSchemaVisitStepContextProvider,
+  ): void => {
+    const directiveWrapped = new DirectiveWrapper(directive);
+    const args = directiveWrapped.getArguments({
+      directiveName,
+      object: parent as ObjectTypeDefinitionNode,
+      field: definition,
+      directive,
+      limit: defaultLimit,
+    } as ManyToManyDirectiveConfiguration);
+
+    validateModelDirective(args);
+    args.connectionFields = [];
+
+    if (!isListType(definition.type)) {
+      throw new InvalidDirectiveError(`@${directiveName} must be used with a list.`);
+    }
+
+    addDirectiveToRelationMap(this.relationMap, args);
```

Comment:
nit: is there a reason `addDirectiveToRelationMap` is not a type method and avoid passing the `this.relationMap`

Reply:
No strong reason it couldn't be one, but my reasoning was:

1. It makes code easier to move around if it's not an instance method.
2. I didn't want to make it part of the public API. Our transpilation process doesn't actually make private things private as far as I can tell.